### PR TITLE
Update Kubernetes dependencies to v1.25.4

### PR DESCRIPTION
**Pull request by RainbowMango**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
This PR updates Kubernetes dependencies to v1.25.4.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
Some changes made at Kubernetes that we must to adopt:
- https://github.com/kubernetes/kubernetes/pull/109303 introduced a `Destroy()` interface to storage
- https://github.com/kubernetes/kubernetes/pull/110088 removed parameters from `Validate()` function

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by RainbowMango** [comment]
/assign @lonelyCZ @XiShanYongYe-Chang @ikaven1024 

Please help to review the second commit which contains all the adoption changes.

**Comment by ikaven1024** [comment]
LGTM

**Review by XiShanYongYe-Chang — COMMENTED** [review]
/lgtm

File: pkg/registry/cluster/storage/proxy.go [hunk 1]
```diff
@@ -58,3 +58,8 @@ func (r *ProxyREST) Connect(ctx context.Context, id string, options runtime.Obje
 	}
 	return proxy.ConnectCluster(ctx, cluster, proxyOpts.Path, secretGetter, responder)
 }
+
+// Destroy cleans up its resources on shutdown.
+// Not implemented.
+func (r *ProxyREST) Destroy() {
```
Comment (XiShanYongYe-Chang) [inline pkg/registry/cluster/storage/proxy.go:64]: // Given that underlying store is shared with REST,
// we don't destroy it here explicitly.

File: pkg/registry/cluster/storage/storage.go [hunk 2]
```diff
@@ -128,3 +128,8 @@ func (r *StatusREST) Update(ctx context.Context, name string, objInfo rest.Updat
 func (r *StatusREST) GetResetFields() map[fieldpath.APIVersion]*fieldpath.Set {
 	return r.store.GetResetFields()
 }
+
+// Destroy cleans up its resources on shutdown.
+// Not implemented.
+func (r *StatusREST) Destroy() {
```
Comment (XiShanYongYe-Chang) [inline pkg/registry/cluster/storage/storage.go:134]: // Given that underlying store is shared with REST,
// we don't destroy it here explicitly.

File: pkg/registry/search/storage/proxy.go [hunk 3]
```diff
@@ -70,3 +70,8 @@ func (r *ProxyingREST) Connect(ctx context.Context, _ string, _ runtime.Object,
 	klog.V(4).Infof("ProxyingREST connect %v", proxyPath)
 	return r.ctl.Connect(ctx, proxyPath, responder)
 }
+
+// Destroy cleans up its resources on shutdown.
+// Not implemented.
+func (r *ProxyingREST) Destroy() {
```
Comment (XiShanYongYe-Chang) [inline pkg/registry/search/storage/proxy.go:76]: // Given no underlying store, we don't destroy anything
// here explicitly.

File: pkg/registry/search/storage/resourceregistry.go [hunk 4]
```diff
@@ -86,3 +86,8 @@ func (r *StatusREST) Update(ctx context.Context, name string, objInfo rest.Updat
 func (r *StatusREST) GetResetFields() map[fieldpath.APIVersion]*fieldpath.Set {
 	return r.store.GetResetFields()
 }
+
+// Destroy cleans up its resources on shutdown.
+// Not implemented.
+func (r *StatusREST) Destroy() {
```
Comment (XiShanYongYe-Chang) [inline pkg/registry/search/storage/resourceregistry.go:92]: // Given that underlying store is shared with REST,
// we don't destroy it here explicitly.

File: pkg/registry/search/storage/search.go [hunk 5]
```diff
@@ -92,3 +92,8 @@ func (r *SearchREST) Connect(ctx context.Context, id string, _ runtime.Object, r
 		return nil, fmt.Errorf("connect with unrecognized search category %s", id)
 	}
 }
+
+// Destroy cleans up its resources on shutdown.
+// Not implemented.
+func (r *SearchREST) Destroy() {
```
Comment (XiShanYongYe-Chang) [inline pkg/registry/search/storage/search.go:98]: // Given no underlying store, we don't destroy anything
// here explicitly.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
I'm sorry, I made some comments after `lgtm`, and I thought we could add some comments as well.

**Comment by RainbowMango** [comment]
That's exactly what I want to say. Thanks. Updated the comments.

**Pull request merged**